### ISSUE TITLE
New invoice order handling with pre-authorization

### ISSUE THREAD

**Issue by skymeissner** [PAYONE-GmbH/magento-2#65]
If you create order with authorization method "pre-autorization" and payment method "invoice" the amount is not yet captured. 
To tell PAYONE to capture the amount I have to invoice and capture online. But that marks my invoice and order immeditely as paid, although no payment has yet occurred.

Can you reproduce the problem? Or do I have a problem in my workflow?

I solved this with interceptor for \Magento\Sales\Model\Order\Invoice::pay().

```
public function aroundPay(\Magento\Sales\Model\Order\Invoice $invoice, callable $proceed)
    {
        $order = $invoice->getOrder();

        // capture online with PAYONE pre-authorization should not immediately mark the invoice as paid
        // paying the invoice (invoice->setForcePay(true)) should
        if ($order->getPayment()->getMethod() == 'payone_invoice'
            && $order->getPayoneAuthmode() === 'preauthorization'
            && !$invoice->getForcePay()
        ) {
            return $invoice;
        }
        return $proceed();
    }
```

Now in the transaction status paid-event one must call $invoice->setForcePay(true)->pay();

Please let me know if this solution is on interest to you then I'll attach a PR.

**Comment by fjbender** [comment]
That's a toughie. In Mage 1, the invoice is marked as "Paid" right after the successful capture as well - which, however, seems counter-intuitive to me.

Still, there might be a good reason why we do it that way - I'll try and find out.

**Comment by fjbender** [comment]
Hi @skymeissner,

Sorry, this issue got kinda lost. I'd be happy if you sent a pull request!

**Comment by hreinberger** [comment]
fixed in #140
